src/DataTypes/Serializations/SerializationUUID.cpp

```diff
@@ -44,10 +45,11 @@ void SerializationUUID::serializeTextQuoted(const IColumn & column, size_t row_n
 
 void SerializationUUID::deserializeTextQuoted(IColumn & column, ReadBuffer & istr, const FormatSettings &) const
 {
+    std::string quoted_string;
+    readQuotedStringWithSQLStyle(quoted_string, istr);
```

Comment:
This looks wrong, SQL style escaping is unneeded for this method.

Reply:
The following deserialization methods are using the same or a similar approach:
```
SerializationAggregateFunction
SerializationEnum
SerializationFixedString
SerializationObject
SerializationString
```

For example, `SerializationString` uses `readQuotedStringInto<true>` which will use `readAnyQuotedStringInto<'\'',true>`.
OTOH, `readQuotedStringWithSQLStyle` which is `readQuotedStringInto<true>`, so basically the same thing.

If Strings are being escaped, why shouldn't strings that represent UUIDs be escaped too?